dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -174,14 +174,29 @@ def just_authorized_google_classroom(user, params)
       scopes.include?('classroom.rosters.readonly')
   end
 
-  def silent_takeover(oauth_user)
+  def silent_takeover(oauth_user, auth_hash)
     # Copy oauth details to primary account
     @user = User.find_by_email_or_hashed_email(oauth_user.email)
-    @user.provider = oauth_user.provider
-    @user.uid = oauth_user.uid
-    @user.oauth_refresh_token = oauth_user.oauth_refresh_token
-    @user.oauth_token = oauth_user.oauth_token
-    @user.oauth_token_expiration = oauth_user.oauth_token_expiration
+    if @user.migrated?
+      AuthenticationOption.new(
```

Comment:
`AuthenticationOption.new(...).save!` can be simplified to `AuthenticationOption.create`. also, if we want to use the `!` operator here, should there be a `begin... rescue` around the entire block so we can safely fail? or will the failure be caught somewhere else?

Reply:
That's a good point, maybe instead of throwing, this should report a meaningful error message in some way.

Reply:
yeah, in that case you could do the:
```
if auth_option.save
  # success case
else
  # error case
end
```
pattern